packages/babel-plugin-proposal-decorators/src/index.ts

```diff
@@ -16,7 +17,7 @@ export default declare((api, options) => {
     throw new Error("'legacy' must be a boolean.");
   }
 
-  const { decoratorsBeforeExport } = options;
+  const { decoratorsBeforeExport, version } = options;
```

Comment:
It seems to me the `decoratorsBeforeExport` option is required only when `version` is `2018-09` (all 2021 tests have it set to `false`). If so we should not throw `decoratorsBeforeExport` is required when version is `2021-12`.